numba/tests/test_runtests.py

```diff
@@ -120,20 +120,33 @@ def get_count(arg_list):
             self.assertEqual(total, included + excluded)
 
     def test_check_slice(self):
-        tmp = self.get_testsuite_listing(['-j','0,5,1'])
-        l = [x for x in tmp if x.startswith('numba.')]
-        self.assertEqual(len(l), 5)
+        tmpAll = self.get_testsuite_listing([])
+        tmp1 = self.get_testsuite_listing(['-j', '0:2'])
+        tmp2 = self.get_testsuite_listing(['-j', '1:2'])
+        lAll = {x for x in tmpAll if x.startswith('numba.')}
+        l1 = {x for x in tmp1 if x.startswith('numba.')}
+        l2 = {x for x in tmp2 if x.startswith('numba.')}
+        # The variance between two adjacent shards should be less than 5% of
+        # the total
```

Comment:
Is this just a guess based on what happens or is there some fundamental property involved here? Also, is this the variance or the difference?

Reply:
It's fundamental. If you have a hash distribution, you don't extend it to be perfectly flat.